types/jest/jest-tests.ts

```diff
@@ -544,6 +544,23 @@ class SpyableClass {
 // $ExpectType SpyInstance<SpyableClass, [number, string]> || SpyInstance<SpyableClass, [a: number, b: string]>
 jest.spyOn({ SpyableClass }, "SpyableClass");
 
+interface SpyableWithIndexSignature {
+    [index: string]: {
+        [x: string]: any;
+    };
+    prop: {some: string};
+    methodOne: () => void;
+}
```

Comment:
```suggestion
    methodTwo: (s: string, b: boolean) => { b: boolean; n: number };
}
```

It would be useful to add another test for a method with arguments and a non-void return value

Reply:
Added in [`c8f0ac1` (#60971)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60971/commits/c8f0ac1e1bbc914e647cb69d7b4b74437fade364), thanks for review!